tooling/sync/generate_command_surfaces.py

```
@@ -0,0 +1,148 @@
+def _extract_tiers(manifest: dict[str, Any]) -> tuple[list[str], list[str]]:
+    must_command_contracts = _require_mapping(manifest, "must_command_contracts")
+    command_tiers = _require_mapping(manifest, "command_tiers")
+
+    invalid_tiers = sorted(
+        command
+        for command, tier in command_tiers.items()
+        if not isinstance(command, str) or tier not in _ALLOWED_TIERS
+    )
+    if invalid_tiers:
+        details = ", ".join(invalid_tiers)
+        raise ValueError(f"command_tiers contains invalid entries: {details}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)